src/components/AddressDetailExplorer.js

```diff
@@ -46,412 +49,362 @@ class AddressDetailExplorer extends React.Component {
-        page: 0,
-        searchAfter: {
-          lastTx: null,
-          lastTs: null,
-        },
+  const [page, setPage] = useState(0);
+  const [selectedToken, setSelectedToken] = useState('');
+  const [balance, setBalance] = useState({});
+  const [transactions, setTransactions] = useState([]);
+  const [hasAfter, setHasAfter] = useState(false);
+  const [hasBefore, setHasBefore] = useState(false);
+  const [loadingSummary, setLoadingSummary] = useState(false);
+  const [loadingHistory, setLoadingHistory] = useState(false);
+  const [loadingTokens, setLoadingTokens] = useState(true);
+  const [loadingPagination, setLoadingPagination] = useState(false);
+  const [errorMessage, setErrorMessage] = useState('');
+  const [warningRefreshPage, setWarningRefreshPage] = useState(false);
+  const [warnMissingTokens, setWarnMissingTokens] = useState(0);
+  const [selectedTokenMetadata, setSelectedTokenMetadata] = useState(null);
+  const [metadataLoaded, setMetadataLoaded] = useState(false);
+  const [addressTokens, setAddressTokens] = useState({});
+  const [txCache, setTxCache] = useState({});
+  const [showReloadDataButton, setShowReloadDataButton] = useState(false);
+  const [showReloadTokenButton, setShowReloadTokenButton] = useState(false);
+  const [pageSearchAfter, setPageSearchAfter] = useState([
+    {
+      page: 0,
+      searchAfter: {
+        lastTx: null,
+        lastTs: null,
       },
-    ],
-  };
-
-  componentDidMount() {
-    // Expects address on URL
-    this.updateAddress(this.props.match.params.address);
-
-    WebSocketHandler.on('network', this.handleWebsocket);
-  }
-
-  componentWillUnmount() {
-    WebSocketHandler.removeListener('network', this.handleWebsocket);
-  }
-
-  componentDidUpdate(prevProps) {
-    if (prevProps.match.params.address !== this.props.match.params.address) {
-      // Address on the URL changed
-      this.pagination.clearOptionalQueryParams();
-      this.updateAddress(this.props.match.params.address);
-      return;
-    }
-
-    const queryParams = this.pagination.obtainQueryParams();
-
-    // Do we have new URL params?
-    if (!isEqual(this.state.queryParams, queryParams)) {
-      if (queryParams.token !== this.state.queryParams.token && queryParams.token !== null) {
-        // User selected a new token, so we must go to the first page (clear queryParams)
-        this.pagination.clearOptionalQueryParams();
-
-        // Need to get newQueryParams because the optional ones were cleared
-        // Update state to set the new selected token on it
-        // If we don't update this state here we might execute a duplicate request
-        const newQueryParams = this.pagination.obtainQueryParams();
-        this.setState({ queryParams: newQueryParams }, () => {
-          this.reloadTokenData(newQueryParams.token);
-        });
-        return;
-      }
-
-      // update the query params state
-      this.setState({ queryParams, loadingHistory: true });
-    }
-  }
+    },
+  ]);
```

Comment:
[note] I have a concern about this pattern because it may cause too much re-renderings and it may cause other issues.

Reply:
React batches state changes made within a single synchronous iteration, so there will be only a single re-render for all those changes ( except on some very specific async situations ).

I found [this article](https://medium.com/swlh/react-state-batch-update-b1b61bd28cd2) that explains it way better than I could, with details, example codes and live tests.